### DISCUSSION TITLE
libsql-client support to allow serverless installation

### DISCUSSION THREAD

**Discussion by andrius** [dani-garcia/vaultwarden#3614] [Ideas]
The docker image is relatively small and it fits into the serverless, for example with the AWS or Scaleway 

The main issue is the SQLITE. But there is a solution:https://turso.tech/, I believe more will come. According to their FAQ (https://docs.turso.tech/reference/client-access/rust-sdk), this crate should be added to the project... https://crates.io/crates/libsql-client

**Comment by BlackDex** [comment]
Vaultwarden uses Diesel as its ORM. A quick look at the crate you mention seems that it doesn't support Diesel.

So, unless there is going to be a compatibility with Diesel, i doubt we will add this because it will add a huge maintenance burden.

**Reply by BlackDex** [reply]
Also, you can use MariaDB/MySQL or PostgreSQL as an alternative.

**Reply by andrius** [reply]
Thank you

Just to test it I ran it on scaleway

![image](https://github.com/dani-garcia/vaultwarden/assets/26776/c6c7bd93-6285-442b-bbcf-ab28d0c3514f)
